src/apps/common/apputils.c

```diff
@@ -739,6 +753,50 @@ char *skip_blanks(char* s)
 	return s;
 }
 
+#if defined(MSVC)
+char* dirname(char* fullpath)
```

Comment:
You might want to use https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/splitpath-s-wsplitpath-s instead